src/Chunk.cpp

```diff
@@ -741,6 +767,29 @@ void cChunk::MoveEntityToNewChunk(cEntity * a_Entity)
 
 
 
+void cChunk::UpdateCanUnload()
+{
+	bool CanUnload =
+	m_LoadedByClient.empty() &&  // The chunk is not used by any client
+	(!IsDirty()) &&              // The chunk is not dirty
+	(m_StayCount == 0) &&        // The chunk is not in a ChunkStay
+	(m_Presence != cpQueued) ;   // The chunk is not queued for loading / generating (otherwise multi-load / multi-gen could occur)
+
+	if ((CanUnload == false) && (m_CanUnload == true))
```

Comment:
It actually does work as expected in C++ (it only doesn't work in C), but it's a style we don't use. If a variable is of a bool type, we just use it as the condition. So yes, please change it to @SphinxC0re 's style.